### PR TITLE
feat(animations): add new TranslateFromRight animation

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-4696

This PR adds new TranslateFromRight animation, and also adds enter animation to BaseModal component overlay item.

### CODE REVIEW

File: packages/x-components/src/components/modals/base-modal.vue [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-  <component :is="animation">
+  <component :is="animation" @beforeEnter="showOverlay = false" @afterEnter="showOverlay = true">
     <div v-if="open" class="x-modal" data-test="modal">
-      <div ref="modal" class="x-modal__content" data-test="modal-content" role="dialog">
+      <div ref="modal" class="x-modal__content x-list" data-test="modal-content" role="dialog">
         <!-- @slot (Required) Modal container content -->
         <slot />
       </div>
       <div
         ref="overlay"
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/components/modals/base-modal.vue:9]: The doc is wrong, it talks about emitting `click:body`, but the event is `click:overlay`.

File: packages/x-components/src/components/modals/base-modal.vue [hunk 2]
```diff
@@ -1,14 +1,15 @@
 <template>
-  <component :is="animation">
+  <component :is="animation" @beforeEnter="showOverlay = false" @afterEnter="showOverlay = true">
```
Comment (javieri-empathy) [inline packages/x-components/src/components/modals/base-modal.vue:2]: Like you said, this needs a rethink in the future. We want to animate the content div, not the `x-modal` one, probably leaving this wrapper div is a trade of that I would happily do

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/modals/base-modal.vue [hunk 3]
```diff
@@ -173,7 +174,14 @@
     &__overlay {
       width: 100%;
       height: 100%;
-      position: absolute;
+      position: fixed;
+      background-color: rgba(0, 0, 0, 0.7);
+      opacity: 0;
+
+      &--visible {
```
Comment (javieri-empathy) [inline packages/x-components/src/components/modals/base-modal.vue:181]: Shouldn't the class be `is-visible`?

File: packages/x-components/src/components/modals/base-modal.vue (hunk 2 above)
Comment (tajespasarela) [reply packages/x-components/src/components/modals/base-modal.vue:2]: That tradeoff breaks my mobile layout.

We will come back to this in other moment

File: packages/x-components/src/components/modals/base-modal.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/components/modals/base-modal.vue:9]: done

I don't know if this was for some corner case or it was just a refactor garbage

File: packages/x-components/src/components/modals/base-modal.vue (hunk 3 above)
Comment (tajespasarela) [reply packages/x-components/src/components/modals/base-modal.vue:181]: you're right. Done

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**